pkg/backend/backend.go

```diff
@@ -356,18 +356,27 @@ func (c *backendClient) GetStackOutputs(ctx context.Context, name string) (resou
 	if s == nil {
 		return nil, fmt.Errorf("unknown stack %q", name)
 	}
-	snap, err := s.Snapshot(ctx, c.secretsProvider)
+
+	secretsProvider := newStackOutputsProvider(c.secretsProvider)
+	snap, err := s.Snapshot(ctx, secretsProvider)
 	if err != nil {
 		return nil, err
 	}
+
 	res, err := stack.GetRootStackResource(snap)
 	if err != nil {
 		return nil, fmt.Errorf("getting root stack resources: %w", err)
 	}
+
+	providerErr := secretsProvider.Error()
```

Comment:
This isn't obvious at first glance, if this stays it needs a comment around what this Error is

Reply:
Reworked as discussed using a callback.